src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,496 @@
+                this.Flush();
+
+                // If there are unprocessed chars, encode and write them as the final block.
+                ReadOnlySpan<char> charsNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, numOfCharsNotWrittenFromPreviousChunk);
+
+                if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    if (!this.jsonWriter.IsWritingJson)
+                    {
+                        int firstIndexToEscape = this.jsonWriter.NeedsEscaping(charsNotProcessedFromPreviousChunk);
+                        this.WriteChunk(charsNotProcessedFromPreviousChunk, charsNotProcessedFromPreviousChunk.Length, firstIndexToEscape, true);
+
+                        // set numOfCharsNotWrittenFromPreviousChunk to 0
+                        this.numOfCharsNotWrittenFromPreviousChunk = 0;
+                    }
+                    else 
+                    {
+                        WriteChunkWithoutEscaping(charsNotProcessedFromPreviousChunk, charsNotProcessedFromPreviousChunk.Length, true);
+                    }
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<char>.Shared.Return(this.buffer);
+                }
+
+                base.Dispose(disposing);
+            }
+
+            /// <summary>
+            /// Asynchronously disposes of the current object and performs cleanup operations.
+            /// </summary>
+            /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+            public override async ValueTask DisposeAsync()
+            {
+                await this.FlushAsync();
+
+                // If there are unprocessed chars, encode and write them as the final block.
+                ReadOnlyMemory<char> charsNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0, numOfCharsNotWrittenFromPreviousChunk);
+
+                if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    if (!this.jsonWriter.IsWritingJson)
+                    {
+                        int firstIndexToEscape = this.jsonWriter.NeedsEscaping(charsNotProcessedFromPreviousChunk.Span);
+                        this.WriteChunk(charsNotProcessedFromPreviousChunk.Span, charsNotProcessedFromPreviousChunk.Length, firstIndexToEscape, true);
+
+                        // set numOfCharsNotWrittenFromPreviousChunk to 0
+                        this.numOfCharsNotWrittenFromPreviousChunk = 0;
+                    }
+                    else 
+                    {
+                        WriteChunkWithoutEscaping(charsNotProcessedFromPreviousChunk.Span, charsNotProcessedFromPreviousChunk.Length, true);
+                    }
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<char>.Shared.Return(this.buffer);
+                }
+
+                await this.jsonWriter.FlushAsync().ConfigureAwait(false);

```

Comment:
Curious why in this case you didn't call `base.DisposeAsync()` but you called it in the sync version. I'm not suggesting that you should call it. I'm just curious to understand why.

Reply:
I think since the base class `TextWriter` implements the `IAsyncDisposable` Interface, it's ok to call `base.DisposeAsync`.  I'll add it.